LoopKit/HealthStoreUnitCache.swift

```diff
@@ -78,6 +88,37 @@ public class HealthStoreUnitCache {
         return unit
     }
 
+    private func updateCachedUnits() {
+        let quantityTypeIdentifiers = unitCache.value.keys
+        for quantityTypeIdentifier in quantityTypeIdentifiers {
+            _ = self.getHealthStoreUnitAndUpdateCache(for: quantityTypeIdentifier)
+        }
+    }
+
+    private func updateCache(for quantityTypeIdentifier: HKQuantityTypeIdentifier, with unit: HKUnit?) {
+        guard unit != unitCache.value[quantityTypeIdentifier] else {
```

Comment:
Consider moving this inside the `mutate` below so there isn't the timing gap between this check and the update.